src/Runner.Worker/Handlers/ScriptHandler.cs

```diff
@@ -259,6 +259,16 @@ public async Task RunAsync(ActionRunStage stage)
 
             // dump out the command
             var fileName = isContainerStepHost ? shellCommand : commandPath;
+#if OS_OSX
+            if (Environment.ContainsKey("DYLD_INSERT_LIBRARIES"))
+            {
+                // launch `node macOSRunInvoker.js shell args` instead of `shell args` to avoid macOS SIP remove `DYLD_INSERT_LIBRARIES` when launch process
+                string node12 = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), "node12", "bin", $"node{IOUtil.ExeExtension}");
+                string macOSRunInvoker = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "macOSRunInvoker.js");
+                arguments = $"\"{macOSRunInvoker.Replace("\"", "\\\"")}\" \"{fileName.Replace("\"", "\\\"")}\" {arguments}";
```

Comment:
`Replace("\"", "\\\"")` is not enough to escape an arbitrary string for the command line. For a unix-like OS, I think you should at least also escape `\` , `$` and ` . 

Isn't there a utility function somewhere for escaping an argument string in an OS specific way?

Reply:
we don't need to escape those since those are special character for bash/sh, the runner launch process directly, so we only need to worry about `"` and ` `(space).

Reply:
Should we pass the command line using an environment variable instead? Then we avoid escaping complexity

Reply:
Alright, I see you have `_proc.StartInfo.UseShellExecute = false` in `ProcessInvoker.cs`. 

In that case you probably still need to worry about `\` . Otherwise an argument containing `\"` (a backslash and a quote) might cause some trouble. After the `Replace` you'd end-up with `\\"`, which I think would be interpreted as an escaped backslash and an unescaped `"`.

Wouldn't it be easier to use `ProcessStartInfo.ArgumentList` instead of `ProcessStartInfo.Arguments` ? That looks more convenient.

Reply:
fixed to using ENV.

Reply:
we don't want to be in a business of parse customer's arg string into arg array. :)